### PR TITLE
add preloader for categories

### PR DESCRIPTION
**Pull request by tiffanyqi**
Added a preloader for categories, which only shows the categories graph and respective actions when categories is fully loaded. Easy to add to other actions as well!

Temporarily used a preloader, we can change it to anything we want later on.

#39

### CODE REVIEW

File: cal/cal/static/js/app.js [hunk 1]
```diff
@@ -111,6 +111,8 @@ analyticsApp.component('tags', {
 analyticsApp.controller('CategoriesCtrl', function($scope, $http){
   var categoryUrl = '/v1/colorcategories';
   $scope.categories = [];
+  var categories = $scope.categories;
+  categories.dataLoaded = false;
```
Comment (MaxFangX) [inline cal/cal/static/js/app.js]: You can just do `$scope.categories.dataLoaded = false;`

File: cal/cal/static/js/app.js [hunk 2]
```diff
@@ -124,6 +126,7 @@ analyticsApp.controller('CategoriesCtrl', function($scope, $http){
           include: true
         });
       }
+      categories.dataLoaded = true;
```
Comment (MaxFangX) [inline cal/cal/static/js/app.js]: same as above comment

File: cal/cal/static/js/app.js [hunk 3]
```diff
@@ -186,7 +189,7 @@ analyticsApp.controller('CategoriesCtrl', function($scope, $http){
   };
 
   // categories pie chart
-  $scope.options = {
+  $scope.categoryPie = {
```
Comment (MaxFangX) [inline cal/cal/static/js/app.js:268]: +1 for the name change

**Comment by tiffanyqi** [comment]
@MaxFangX fixed your comments! How's it looking now?

**Review by MaxFangX — APPROVED** [review]
+2

**Pull request merged**